iceoryx_posh/test/moduletests/test_roudi_process.cpp

```diff
@@ -0,0 +1,122 @@
+#include "iceoryx_posh/internal/roudi/roudi_process.hpp"
+#include "iceoryx_posh/mepoo/mepoo_config.hpp"
+#include "iceoryx_posh/roudi/memory/roudi_memory_interface.hpp"
+#include "iceoryx_posh/version/compatibility_check_level.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+#undef protected
+#undef private
+
+using namespace ::testing;
+using namespace iox::roudi;
+using namespace iox::popo;
+using namespace iox::runtime;
+using ::testing::Return;
+
+/*class MqInterfaceUser_Mock : public iox::runtime::MqInterfaceUser
+{
+  public:
+    MqInterfaceUser_Mock()
+        : iox::runtime::MqInterfaceUser("MockTest", 5L, 512L)
+    {
+    }
+    MOCK_METHOD1(send, bool(MqMessage));
+};*/
+class MqInterfaceUser_Mock : public iox::roudi::RouDiProcess
+{
+  public:
+    MqInterfaceUser_Mock()
+        : iox::roudi::RouDiProcess("TestRoudiProcess", 200, nullptr, true, 0x654321, 255)
+    {
+    }
+    MOCK_METHOD1(sendToMQ, void(MqMessage));
+};
+
+class RouDiProcess_test : public Test
+{
+  public:
+    const iox::ProcessName_t processname = "TestRoudiProcess";
+    int32_t pid = 200;
+    iox::mepoo::MemoryManager* payloadMemoryManager{nullptr};
+    bool isMonitored = true;
+    const uint64_t payloadSegmentId = 0x654321;
+    const uint64_t sessionId = 255;
+    MqInterfaceUser_Mock mqIntrfceusermock;
+};
+
+TEST_F(RouDiProcess_test, CTor)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.getPid(), Eq(200));
+}
+
+TEST_F(RouDiProcess_test, getPid)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.getPid(), Eq(200));
+}
+
+TEST_F(RouDiProcess_test, getName)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.getName(), Eq(std::string("TestRoudiProcess")));
+}
+
+TEST_F(RouDiProcess_test, isMonitored)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.isMonitored(), Eq(true));
+}
+
+TEST_F(RouDiProcess_test, getPayloadSegId)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.getPayloadSegmentId(), Eq(0x654321));
+}
+
+TEST_F(RouDiProcess_test, getSessionId)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.getSessionId(), Eq(255));
+}
+
+TEST_F(RouDiProcess_test, getPayloadMemoryManager)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.getPayloadMemoryManager(), Eq(nullptr));
+}
+
+// Not able to test sendToMQ as there is no return value and mocking is not currently done
+
+TEST_F(RouDiProcess_test, sendToMQ)
```

Comment:
What about tests when send is failing?

Reply:
sendToMQ function is an wrapper over MqBase::send, error handling is done in MqBase::send not inside the wrapper. Already test cases are available to test failure conditions of MqBase::send

Reply:
That's not the whole truth, in `MqBase::send` are only logs printed when the message is not valid. In the underlying mq_send is a smart_c wrapper which also do some checks. 
But this does not solve the problem here, because the question is: What do we do in RouDi Process Class when the message queue is not sending? 
`m_mq.send(data);` is returning a bool which is not handled by RouDiProcess, thats not good. This function is often used by the ProcessManager and we should react when a message is not transferred!

I see that there is already the Errorhandler used, this can be here maybe also the case.
@chiranjeevimaddi Can you make a proposal what to do here? The [Error Handling Guide](https://github.com/eclipse/iceoryx/blob/master/doc/design/error-handling.md) maybe helps you.

Reply:
Regarding the failed condition of MqBase::send, i am thinking of using errorHandler(Error::kSOME_ERROR_CODE, nullptr, ErrorLevel::SEVERE); 
ErrorCode: i need to verify if i have one or i need to create one. 
@dkroenke can you comment on my opinion?

Reply:
@chiranjeevimaddi You can take a look at line 293 in roudi_process.cpp for an example. The errorhandler soes not need to be instantiated.

Reply:
Done